### PR TITLE
Wrong pin set for PIN_APA102_DATA for FeatherS2 board.h

### PR DESCRIPTION
**Pull request by UnexpectedMaker**
Could be the cause of the "invalid header: 0x0a324655" issue in the automatic UF2 build.

### CODE REVIEW

**Comment by UnexpectedMaker** [comment]
I should have checked this out into a new branch to seperate it into a new PR, sorry @tannewt - but here it is... APA support, for at least board.c
I'm Stil not sure what the LED support in the boards_start.c is for... Espressif level bootloader? Does that ever get hit?

File: components/boards/boards.c [hunk 1]
```diff
@@ -147,6 +167,25 @@ void led_blinky_cb(TimerHandle_t xTimer)
 }
 #endif
 
+#ifdef PIN_APA102_SCK
+TimerHandle_t blinky_tm = NULL;
+
+void led_blinky_cb(TimerHandle_t xTimer)
```
Comment (hathach) [inline components/boards/boards.c:173]: should be merge with above led_blinky_cb(), see below note for suggestion.

File: components/boards/boards.c [hunk 2]
```diff
@@ -177,4 +216,33 @@ void board_led_state(uint32_t state)
     break;
   }
   #endif
-}
+
```
Comment (hathach) [inline components/boards/boards.c:219]: this code is exactly the same as above switch, can you merge them together you can try to `define rgb_set()` that expand to neopixel_set or setAPA depending on the PIN_xxx.

**Review by hathach — COMMENTED** [review]
the change look good, though it would be perfect if you could merge the current code per review

File: components/boards/boards.c (hunk 2 above)
Comment (UnexpectedMaker) [reply components/boards/boards.c:219]: Sure, I was thinking of that when I implemented it, but wasn't sure if it was ok to change the neopixel stuff as that wasn't my code ;)

**Comment by hathach** [comment]
> I should have checked this out into a new branch to seperate it into a new PR, sorry @tannewt - but here it is... APA support, for at least board.c
> I'm Stil not sure what the LED support in the boards_start.c is for... Espressif level bootloader? Does that ever get hit?

2nd stage bootloader indicator is implemented separately here 
https://github.com/adafruit/uf2-esp32s/blob/master/components/bootloader/subproject/main/bootloader_start.c#L253
The reason is 2nd bootloader can only access to only/mostly rom API for peripherals since most peripherial library relies on freeRTOS and memory region that aren't initialized yet in 2nd bootloader. I have to bit-banging neopixel in above link. You could try to link the led_strip_spi_apa102 to bootloader to see if it works out
https://github.com/adafruit/uf2-esp32s/blob/master/components/bootloader/subproject/main/CMakeLists.txt

**Comment by UnexpectedMaker** [comment]
@hathach I've updated the `boards.c` using a wrapper called `set_pixel()` and neatened it all up.
I tried (quickly) to get my APA stuff implemented into 'bootloader-start.c` but I could't seem to get the CMake include search to find the files. I'll try again over the weekend, and if cant get it working, I might try to just bit bang CLK/DATA instead.

**Comment by UnexpectedMaker** [comment]
Hmm, didn't see the build error - think I just fixed it. Well, it'c compiling here now successfully for both soala and featherS2.

File: components/led_strip_spi_apa102/include/led_strip_spi_apa102.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+#include <stdio.h>
+#include <string.h>
+#include <math.h>
+#include <stdlib.h>
+
+// #include "freertos/FreeRTOS.h"
+// #include "freertos/task.h"
+// #include "esp_system.h"
+// #include "driver/gpio.h"
```
Comment (hathach) [inline components/led_strip_spi_apa102/include/led_strip_spi_apa102.h:12]: please remove these header if you don't need it

**Review by hathach — APPROVED** [review]
look great now, and ready for merging. Though will you update the 2nd stage bootloader to support the rgb as well, or you want to it in a separate PR.

File: components/led_strip_spi_apa102/include/led_strip_spi_apa102.h (hunk 3 above)
Comment (UnexpectedMaker) [reply components/led_strip_spi_apa102/include/led_strip_spi_apa102.h]: I removed the commented out headers, sorry.
Yeah I'm going to do the second stage bootloader as a seperate PR :)

**Review by hathach — APPROVED** [review]
it is perfect now.

**Pull request merged**